### PR TITLE
Google pub/sub master

### PR DESCRIPTION
**Pull request by ruchiramoitra**

### CODE REVIEW

File: v0/destinations/googlepubsub/transform.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const { eventToTopicMap } = event.destination.Config;
+  const hashMap = getHashFromArray(eventToTopicMap, "from", "to");
+  return hashMap["*"] || hashMap[message.type] || hashMap[message.event];
+}
+
+function process(event) {
+  const topicPath = getTopic(event);
+  if (topicPath) {
+    const splitPath = topicPath.split("/");
+    const topicId = splitPath[3];
```
Comment (prabrishac) [inline v0/destinations/googlepubsub/transform.js:14]: check `splitPath` length before accessing 3rd indexed element.

Comment (prabrishac) [reply v0/destinations/googlepubsub/transform.js:14]: and handle error.

File: v0/destinations/googlepubsub/transform.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    console.log({
```
Comment (prabrishac) [inline v0/destinations/googlepubsub/transform.js:15]: remove console.log

**Review by prabrishac — CHANGES_REQUESTED** [review]

**Comment by prabrishac** [comment]
minor changes requested.

**Review by prabrishac — APPROVED** [review]

**Review by arnab-p — APPROVED** [review]

**Pull request merged**